homeassistant/components/drop/sensor.py

```diff
@@ -0,0 +1,501 @@
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "current_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the current flow rate sensor."""
+        super().__init__("current_flow_rate", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current flow rate."""
+        if self._device.current_flow_rate is None:
+            return None
+        return round(self._device.current_flow_rate, 1)
+
+
+class DROP_PeakFlowRateSensor(DROP_Entity, SensorEntity):
+    """Monitors the peak water flow rate for the day."""
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "peak_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the peak flow rate sensor."""
+        super().__init__("peak_flow_rate", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current flow rate."""
+        if self._device.peak_flow_rate is None:
+            return None
+        return round(self._device.peak_flow_rate, 1)
+
+
+class DROP_WaterUsedTodaySensor(DROP_Entity, SensorEntity):
+    """Monitors the total water used for the day."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "water_used_today"
+
+    def __init__(self, device) -> None:
+        """Initialize the water used today sensor."""
+        super().__init__("water_used_today", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the total used today."""
+        if self._device.water_used_today is None:
+            return None
+        return round(self._device.water_used_today, 1)
+
+
+class DROP_AverageWaterUsedSensor(DROP_Entity, SensorEntity):
+    """Monitors the average water used over the last 90 days."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "average_water_used"
+
+    def __init__(self, device) -> None:
+        """Initialize the average water used sensor."""
+        super().__init__("average_water_used", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the average used over the last 90 days."""
+        if self._device.average_water_used is None:
+            return None
+        return round(self._device.average_water_used, 1)
+
+
+class DROP_CapacityRemainingSensor(DROP_Entity, SensorEntity):
+    """Monitors the soft water capacity remaining on a softener."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
```

Comment:
```suggestion
    _attr_device_class= SensorDeviceClass.WATER
    _attr_native_unit_of_measurement = UnitOfVolume.GALLONS
```

Please use the enum values here, and best is to set a device class where applicable and not set a custom icon.

Reply:
This applies to all other sensors in this file as well.

Reply:
Good call; I patterned my first sensor off of an example that doesn't have a SensorDeviceClass element and then ran with it.  I set the device class where applicable in f670c323d74e2ce21a05f08b0824c637ec257cf4.